### PR TITLE
rsa-pss signatures

### LINKED ISSUE #275: type of key is not used for signature algorithm negotiation
When the client advertises only `rsa_pss_pss_*` signatures and the server has a rsaEncryption key, it will still sign the Certificate Verify with those algorithms, similarly, if client advertises `rsa_pss_rsae_*` signatures and the server has a rsassa-pss key, it will sign the Certificate Verify with those schemes

the correct behaviour is to abort the connection with `handshake_failure` instead of sending ServerHello message

### PR DESCRIPTION
**Pull request by tomato42**
TLS1.3 changed in that there are different signature schemes based on whether the signing key (and thus associated certificate) is of a rsa-pss key or of a rsaEncryption key.

That means, when creating signature, we need to select the correct algorithm for the ServerKeyExchange or CertificateVerify.

tests were passing because #279 is not fixed

fixes #275 

---
This change is [Reviewable](https://reviewable.io/reviews/tomato42/tlslite-ng/311)

### CODE REVIEW

File: tlslite/tlsconnection.py [hunk 1]
```diff
@@ -3649,12 +3649,14 @@ def _sigHashesToList(settings, privateKey=None, certList=None,
                         continue
                     # advertise support for both rsaEncryption and RSA-PSS OID
                     # key type
-                    sigAlgs.append(getattr(SignatureScheme,
-                                           "rsa_{0}_rsae_{1}"
-                                           .format(schemeName, hashName)))
-                    sigAlgs.append(getattr(SignatureScheme,
-                                           "rsa_{0}_pss_{1}"
-                                           .format(schemeName, hashName)))
+                    if certType != 'rsa-pss':
```
Comment (nmav) [inline tlslite/tlsconnection.py]: Looks good, though the loop does look it assumes that RSA is the only thing that is supported.

File: unit_tests/test_tlslite_constants.py [hunk 2]
```diff
@@ -219,3 +219,13 @@ def test_getHash_with_valid_new_name(self):
...
 
         self.assertEqual(ret, 'sha384')
+
+    def test_getKeyType_with_valid_new_name(self):
+        ret = SignatureScheme.getKeyType('rsa_pss_pss_sha384')
+
+        self.assertEqual(ret, 'rsa')
+
+    def test_getPadding_with_valid_new_name(self):
+        ret = SignatureScheme.getPadding('rsa_pss_pss_sha256')
```
Comment (nmav) [inline unit_tests/test_tlslite_constants.py]: Not sure how complete this should be rsa_pss_rsae_sha384 and 512 seem to be missing. The rsa_pss_sha512 is confusing in the presence of the other options.

**Comment by nmav** [comment]
Other than the comments lgtm

File: tlslite/tlsconnection.py (hunk 1 above)
Comment (tomato42) [reply tlslite/tlsconnection.py]: yes, it does do that, but it's expected as ECDSA is not supported

**Comment by tomato42** [comment]
merged manually as 1b6ec55

**Pull request closed without merging**